core/src/main/java/io/questdb/griffin/engine/functions/groupby/ApproxPercentileDoubleGroupByFunction.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.ArrayColumnTypes;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.map.MapValue;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.engine.functions.*;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import io.questdb.std.histogram.org.HdrHistogram.DoubleHistogram;
+
+public class ApproxPercentileDoubleGroupByFunction extends DoubleFunction implements GroupByFunction, BinaryFunction {
+    // specifies the precision for the recorded values (between 0 and 5).
+    // trade-off between memory usage and accuracy.
+    private final int numberOfSignificantValueDigits = 3;
```

Comment:
Yeah I was considering this too. To have it as an optional 3rd argument, do we have to create a new `FunctionFactory` with a different signature, or is there another way?

Reply:
There are two options:
* One more `FunctionFactory` with `approx_percentile(DDi)` signature.
* Single  `FunctionFactory` with `approx_percentile(DV)`. Here `V` stands for variable list of args. The downside is that you'd have to check the number of arguments and their types in the factory code (check `RndStringListFunctionFactory` as an example of how this can be done).

Reply:
I see, I'll go with the first one as I think it's cleaner.

Reply:
And just to confirm, the lowercase `i` in `DDi` means it can only be a constant right, not a runtime constant?

Reply:
Yes, a lower-case letter stands for a constant. I don't think anyone would want to use a bind var for the precision.